include/ic_components.h

```diff
@@ -8,38 +8,29 @@
 #include <vector>
 
 namespace IC {
-    class GameObjectComponent {
+    class Component {
     public:
-        GameObjectComponent();
-        ~GameObjectComponent();
+        Component();
+        virtual ~Component() = 0;
```

Comment:
Does this compile? I didn't know you could mark the dtor as a pure virtual function